src/all/mitaku/src/eu/kanade/tachiyomi/extension/all/mitaku/Mitaku.kt

```diff
@@ -0,0 +1,155 @@
+package eu.kanade.tachiyomi.extension.all.mitaku
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+
+class Mitaku : ParsedHttpSource() {
+    override val name = "Mitaku"
+
+    override val baseUrl = "https://mitaku.net"
+
+    override val lang = "all"
+
+    override val supportsLatest = false
+
+    // ============================== Popular ===============================
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/category/ero-cosplay/page/$page", headers)
+
+    override fun popularMangaSelector() = "div.article-container article"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        // Extract and set the URL without domain
+
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+
+        // Extract and set the title
+        title = element.selectFirst("a")!!.attr("title")
+
+        // Extract and set the thumbnail URL
+        thumbnail_url = element.selectFirst("img")?.absUrl("src")
+    }
+
+    override fun popularMangaNextPageSelector() = "div.wp-pagenavi a.page.larger"
+
+    // =============================== Latest ===============================
+    override fun latestUpdatesRequest(page: Int): Request {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesSelector(): String {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesFromElement(element: Element): SManga {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesNextPageSelector(): String? {
+        throw UnsupportedOperationException()
+    }
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val filterList = if (filters.isEmpty()) getFilterList() else filters
+        val tagFilter = filterList.findInstance<TagFilter>()!!
+        val categoryFilter = filterList.findInstance<CategoryFilter>()!!
+        return when {
+            query.isEmpty() && categoryFilter.state != 0 -> GET("$baseUrl/category/${categoryFilter.toUriPart()}/page/$page/")
+            query.isEmpty() && tagFilter.state.isNotEmpty() -> GET("$baseUrl/tag/${tagFilter.state}/page/$page/")
+            query.isNotEmpty() -> GET("$baseUrl/page/$page/?s=$query")
```

Comment:
Use a `HttpUrl.Builder` to properly escape the query. Search for `toHttpUrl().newBuilder()` for examples.